crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -0,0 +1,988 @@
+use crate::{
+    connector::utils::{self, PaymentsAuthorizeRequestData, RefundsRequestData, RouterData},
+    consts,
+    core::errors,
+    types::{self, api, storage::enums},
+};
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PaymentInput {
+    payment_method_id: String,
+    transaction: TransactionBody,
+}
+
+#[derive(Debug, Serialize)]
+pub struct VariablePaymentInput {
+    input: PaymentInput,
+}
+
+#[derive(Debug, Serialize)]
+pub struct BraintreePaymentsRequest {
+    query: String,
+    variables: VariablePaymentInput,
+}
+
+#[derive(Debug, Serialize)]
+pub struct TransactionBody {
+    amount: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for BraintreePaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let query = match item.request.is_auto_capture()?{
+            true => "mutation ChargeCreditCard($input: ChargeCreditCardInput!) { chargeCreditCard(input: $input) { transaction { id legacyId createdAt amount { value currencyCode } status } } }".to_string(),
+            false => "mutation authorizeCreditCard($input: AuthorizeCreditCardInput!) { authorizeCreditCard(input: $input) {  transaction { id legacyId amount { value currencyCode } status } } }".to_string(),
+        };
+        Ok(Self {
+            query,
+            variables: VariablePaymentInput {
+                input: PaymentInput {
+                    payment_method_id: item.get_payment_method_token()?,
+                    transaction: TransactionBody {
+                        amount: utils::to_currency_base_unit(
+                            item.request.amount,
+                            item.request.currency,
+                        )?,
+                    },
+                },
+            },
+        })
+    }
+}
+
+#[derive(Default, Debug, Clone, Deserialize)]
+pub struct BraintreeAuthResponse {
+    data: Option<DataAuthResponse>,
+    errors: Option<Vec<ErrorDetails>>,
+}
+
+#[derive(Default, Debug, Clone, Deserialize)]
+pub struct TransactionAuthChargeResponseBody {
+    id: String,
+    status: BraintreePaymentStatus,
+}
+
+#[derive(Default, Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DataAuthResponse {
+    authorize_credit_card: Option<AuthChargeCreditCard>,
+}
+
+#[derive(Default, Debug, Clone, Deserialize)]
+pub struct AuthChargeCreditCard {
+    transaction: Option<TransactionAuthChargeResponseBody>,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, BraintreeAuthResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, BraintreeAuthResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        if item.response.errors.is_some() {
```

Comment:
replace this with match statement instead of if and else

Reply:
Replaced if else statement with match statement